pennylane/ops/op_math/controlled_decompositions.py

```diff
@@ -16,18 +16,20 @@
 """
 
 from copy import copy
-from typing import Tuple
 
 import numpy as np
 import numpy.linalg as npl
 
 import pennylane as qml
 from pennylane import math
 from pennylane.operation import Operation, Operator
+from pennylane.ops.op_math.decompositions.single_qubit_unitary import (
+    _get_single_qubit_rot_angles_via_matrix,
```

Comment:
Given this is a short private function, I think it would probably make maintenance easier to copy and paste it here, instead of introducing a coupling between the two different files.

This will allow the private function to be updated in each file to suit local needs independently of the other file.

Reply:
I agree. The problem is that `_get_single_qubit_rot_angles_via_matrix` also uses the `_zyz_get_rotation_angles` function from single_qubit_unitary.py file, so the coupling would still be there unless I also copy this function. What do yout think?